qiwis.py

```diff
@@ -145,28 +145,32 @@ class MdiArea(QMdiArea):
     """QMdiArea for the central widget.
     
     Attributes:
-        background: The QPixmap instance of the background image.
+        background_path, background_color: See the predefined constants in _read_config_file().
     """
 
-    def __init__(self, background_path: Optional[str] = None):
+    def __init__(self, background_path: Optional[str], background_color: Optional[str]):
         """Extended.
         
         Args:
-            background_path: The path of the background image.
+            See the attributes section.
         """
         super().__init__()
-        self.background = QPixmap(background_path)
+        self.background_path = background_path
+        self.background_color = background_color
 
-    def paintEvent(self, event: QPaintEvent):
-        """Extended.
+    def paintEvent(self, _event: QPaintEvent):
+        """Overridden.
         
-        Draws the background image.
+        Paints its background.
         """
-        QMdiArea.paintEvent(self, event)
         painter = QPainter(self.viewport())
-        x = (self.width() - self.background.width()) // 2
-        y = (self.height() - self.background.height()) // 2
-        painter.drawPixmap(x, y, self.background)
+        if self.background_color is not None:
+            painter.fillRect(self.rect(), QColor(int(self.background_color, 16)))
+        if self.background_path is not None:
+            image = QPixmap(self.background_path)

```

Comment:
Is this efficient enough? It looks like it loads the image for every paint event.

Reply:
Oh, you're right! This method is called whenever the size of `QMdiArea` is changed.
I will improve it not to load the image every time.